instrumentation/jmx-metrics/library/src/main/java/io/opentelemetry/instrumentation/jmx/yaml/MetricStructure.java

```diff
@@ -59,9 +60,14 @@ abstract class MetricStructure {
   void setType(String t) {
     // Do not complain about case variations
     t = t.trim().toUpperCase(Locale.ROOT);
-    this.metricType = MetricInfo.Type.valueOf(t);
+    try {
+      this.metricType = MetricInfo.Type.valueOf(t);
+    } catch (IllegalArgumentException e) {
+      throw new IllegalArgumentException("Invalid metric type: " + t, e);
+    }
```

Comment:
[for reviewer] this allows to make the error message nicer for the end user when type is invalid.